Sultan_Alsamaani/week_01/day_03/main.js

```diff
@@ -0,0 +1,38 @@
+function squareNumber(number) {
+    return number*number;
+}
+
+function halfNumber(number) {
+    return number/2;
+}
+
+function percentOf(num1,num2) {
+    if (num1>num2) {
+        return (num2/num1)*100+"%";
+    } else {
+        return (num1/num2)*100+"%";
+    }
+}
+function areaOfCircle(num) {
+    return 3.14159*squareNumber(num);
+}
+
+function part2(num) {
+    var result = halfNumber(num);
+    result = squareNumber(result);
+    result = areaOfCircle(result);
+    result = percentOf(result,squareNumber(result));
```

Comment:
This is great! The only problem I see is that you're overriding the value of `result` every time you call the next function.

Reply:
'overriding the value of result'
it similar like this code
var halfNumberResult = halfNumber(num);
 var   squareNumberResult = squareNumber(halfNumberResult);
 var   areaOfCircleResult = areaOfCircle(squareNumberResult);
  var  FinalResult = percentOf(result, squareNumber(areaOfCircleResult));

or in another way
var result =percentOf(areaOfCircle(squareNumber(halfNumber(num))))

advice me please how should i do it
thanks